### PR TITLE
Add downsampling notes to plot documentation

### PR DESCRIPTION
**Pull request by defunctzombie**
The primary motivation for this change is to introduce some user-relevant details about our downsampling strategies on the plot panel. These are added to a new `Downsampling` section.

While working on the docs I felt the structure for the settings and flow between sections was a bit unclear so I moved some things into specific settings and updated some outdated language.

### CODE REVIEW

File: docs/3-visualization/4-panels/plot.md [hunk 1]
```diff
@@ -131,12 +135,30 @@ Note that `.@derivative` will not work with scatter plots (i.e. message paths th
...
 
 ![plot with reference value](/img/docs/visualization/panels/plot/reference-plot.webp)
 
+## Downsampling
+
+To improve performance, the Plot panel will downsample data when points are too dense to meaningfully display.
+
+### Timestamp mode
+
+In timestamp mode, the Plot panel will downsample data using a min/max/first/last downsampling algorithm. This downsampling will preserve extremes so even when downsampled, the plot will still show the highest and lowest values for each time period. The points kept during downsampling will always be from the original data, downsampling does not interpolate or create new points.
```
Comment (snosenzo) [inline docs/3-visualization/4-panels/plot.md:150]: Is "Timestamp mode" something that the user would intuitively know about? Should we change this to say "When using a timestamp for the x-axis" or "when plotting a timeseries". 

Timestamp mode feels over-specific to the degree that I feel like I would've missed a setting above, when in reality it just means you're plotting a timeseries

File: docs/3-visualization/4-panels/plot.md [hunk 2]
```diff
@@ -131,12 +135,30 @@ Note that `.@derivative` will not work with scatter plots (i.e. message paths th
...
+
+When data is downsampled the plot display will automatically hide the "dots" for your data and only show lines. You can zoom until you see dots indicating you are seeing un-downsampled original data.
+
+### Path modes (Scatterplot)
```
Comment (snosenzo) [inline docs/3-visualization/4-panels/plot.md:154]: I feel similarly about the use of `mode` here. Maybe these sections should be titled "Plotting against time" and "Plotting against message values"?

File: docs/3-visualization/4-panels/plot.md (hunk 1 above)
Comment (snosenzo) [reply docs/3-visualization/4-panels/plot.md:150]: Would this section also include when using indices on the x-axis?

Comment (defunctzombie) [reply docs/3-visualization/4-panels/plot.md:150]: These "modes" are mentioned here: https://d4722b64.foxglove-docs.pages.dev/docs/visualization/panels/plot/#x-axis

I could not think of a better word to refer to them. I could say "When using" but that felt like a lot for the section title - are you suggesting that be the first sentence?

File: docs/3-visualization/4-panels/plot.md (hunk 2 above)
Comment (defunctzombie) [reply docs/3-visualization/4-panels/plot.md:154]: Whatever we decide I think it should match up somewhat closely to the docs here: https://d4722b64.foxglove-docs.pages.dev/docs/visualization/panels/plot/#x-axis

**Comment by defunctzombie** [comment]
Any other comments on these changes? I have not come up with a better term than "mode" and generally feel it is semantically accurate enough that readers will understand the intent. The pedantic naming would be "X-axis value setting" but that seems more confusing.

File: docs/3-visualization/4-panels/plot.md (hunk 1 above)
Comment (jtbandes) [inline docs/3-visualization/4-panels/plot.md:150]:
```suggestion
In timestamp mode, the Plot panel will downsample data using a min/max/first/last downsampling algorithm. This downsampling will preserve extremes so even when downsampled, the plot will still show the highest and lowest values for each time period. The points kept during downsampling will always be from the original data — downsampling does not interpolate or create new points.
```

File: docs/3-visualization/4-panels/plot.md [hunk 3]
```diff
@@ -131,12 +135,30 @@ Note that `.@derivative` will not work with scatter plots (i.e. message paths th
...
 
+## Downsampling
+
+To improve performance, the Plot panel will downsample data when points are too dense to meaningfully display.
+
+### Timestamp mode
+
+In timestamp mode, the Plot panel will downsample data using a min/max/first/last downsampling algorithm. This downsampling will preserve extremes so even when downsampled, the plot will still show the highest and lowest values for each time period. The points kept during downsampling will always be from the original data, downsampling does not interpolate or create new points.
+
+When data is downsampled the plot display will automatically hide the "dots" for your data and only show lines. You can zoom until you see dots indicating you are seeing un-downsampled original data.
```
Comment (jtbandes) [inline docs/3-visualization/4-panels/plot.md:152]:
```suggestion
When data is downsampled, the Plot panel will automatically hide the dots for your data and only show lines. You can zoom until you see dots indicating you are seeing the full original data.
```

File: docs/3-visualization/4-panels/plot.md [hunk 4]
```diff
@@ -131,12 +135,30 @@ Note that `.@derivative` will not work with scatter plots (i.e. message paths th
...
+
+### Path modes (Scatterplot)
+
+In path modes, the Plot panel will use different downsampling approaches depending on the "Show lines" setting for the series.
+
+If the series does not have lines enabled, downsampling will remove points which occur at the same pixel location as other points. The rendered plot will look the same as one with no downsampling.
+
+If the series has lines enabled, downsampling will preserve the shape of the data and only downsample datums which would not alter the shape. When data is downsampled the plot display will automatically hide the "dots" for your data and only show lines. You can zoom until you see dots indicating you are seeing un-downsampled original data.
```
Comment (jtbandes) [inline docs/3-visualization/4-panels/plot.md:160]:
```suggestion
If the series has lines enabled, downsampling will preserve the shape of the data and only remove points which would not alter the shape. When data is downsampled, the Plot panel will automatically hide the dots for your data and only show lines. You can zoom until you see dots indicating you are seeing the full original data.
```

**Review by jtbandes — APPROVED** [review]

**Pull request merged**